lib/howl/commands/app_commands.moon

```diff
@@ -302,36 +313,54 @@ command.register
 -----------------------------------------------------------------------
 
 launch_cmd = (working_directory, cmd) ->
-  shell = howl.sys.env.SHELL or '/bin/sh'
-  p = Process {
-    :cmd,
-    :shell,
-    read_stdout: true,
-    read_stderr: true,
-    working_directory: working_directory,
-  }
-
-  buffer = ProcessBuffer p
-  editor = app\add_buffer buffer
-  editor.cursor\eof!
-  buffer\pump!
+  dispatch.launch ->
```

Comment:
Ok, then. There are still some implementation details that need to be worked out. For e.g. the command line currently runs both interactions and commands. (Perhaps it is really designed for interactions - i.e. it blocks - which is why it requires commands to use dispatch.launch). 

With commands specifying both - the `handler` and an optional `interactor` - maybe it's best if the command module itself runs the command handlers, and only invokes the command_line to run the interactor if provided. Also, the `command_line` only runs interactions (and not 'activities' as it does currently - which is used to mean both commands and interactions). This removes one extra term from the terminology (activity) and decouples the commands from the tightly coupled interactions+command_line. I feel this will be a really good change, assuming it's doable without running into other issues.